File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.java
import java.util.*;

class Solution {
    public int solution(String[] babbling) {
        
        int answer = 0;
        String [] words = {"aya", "ye","woo","ma"};
        
        for(String bab : babbling){
            
            for(int i = 0; i <words.length; i++){
                bab = bab.replace(words[i],Integer.toString(i+1));
            }
            
            char temp = '0';
            boolean check = true;
            
            for(int i = 0; i<bab.length(); i++){
                if(Character.isDigit(bab.charAt(i))){
                    if(bab.charAt(i) == temp){
                        check = false;
                        break;
                    }
                    else{
                        temp = bab.charAt(i);
                    }
                    
                }
                else{
                    check = false;
                    break;
                }
            }
            if(check){
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma"]
    
    for bab in babbling:
        for i, word in enumerate(words):
            bab = bab.replace(word, str(i + 1))
        
        temp = '0'
        check = True
        
        for char in bab:
            if char.isdigit():
                if char == temp:
                    check = False
                    break
                else:
                    temp = char
            else:
                check = False
                break
        
        if check:
            answer += 1

    return answer